spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/SpannerConverters.java

```diff
@@ -273,4 +257,46 @@ public ByteArray convert(byte[] bytes) {
 					SPANNER_TO_JAVA_SQL_DATE_CONVERTER,
 					TIMESTAMP_LOCAL_DATE_CONVERTER,
 					TIMESTAMP_LOCAL_DATE_TIME_CONVERTER));
+
+	/**
+	 * An utility function to convert a value inherited from {@link java.util.Date} to the {@link Timestamp}.
+	 * It also can be used to convert a {@link java.sql.Date} value.
+	 * @param date the value to convert
+	 * @param <D> the type of the value to convert
+	 * @return the equivalent Timestamp value
+	 */
+	public static <D extends java.util.Date> Timestamp toTimestamp(@NonNull D date) {
+		long time = date.getTime();
+		long secs = Math.floorDiv(time, 1000L);
+		int nanos = Math.toIntExact((time - secs * 1000L) * 1000000L);
+		return Timestamp.ofTimeSecondsAndNanos(secs, nanos);
+	}
+
+	/**
+	 * An utility function to convert a {@link java.sql.Timestamp} value to the {@link Timestamp}.
+	 * @param timestamp the value to convert
+	 * @return the equivalent Timestamp value
+	 */
+	public static Timestamp toTimestamp(@NonNull java.sql.Timestamp timestamp) {
+		long secs = Math.floorDiv(timestamp.getTime(), 1000L);
+		return Timestamp.ofTimeSecondsAndNanos(secs, timestamp.getNanos());
+	}
+
+	/**
+	 * An utility function to convert a {@link LocalDateTime} value to the {@link Timestamp}.
+	 * @param dateTime the value to convert
+	 * @return the equivalent Timestamp value
+	 */
+	public static Timestamp toTimestamp(@NonNull LocalDateTime dateTime) {
+		return toTimestamp(java.sql.Timestamp.valueOf(dateTime));
+	}
+
+	/**
+	 * An utility function to convert an {@link Instant} value to the {@link Timestamp}.
+	 * @param instant the value to convert
+	 * @return the equivalent Timestamp value
+	 */
+	public static Timestamp toTimestamp(@NonNull Instant instant) {
```

Comment:
If any of those methods are used only once, please just inline them.

Reply:
well, I had a hope to re-use as `public static <D extends java.util.Date> Timestamp toTimestamp(@NonNull D date)` for a sql-data converter in my project :)

Can we move them to `ConversionUtils` class?